code/tests/generic_343.cpp

```diff
@@ -0,0 +1,255 @@
+/*
+Reproducing fstest generic/343
+
+Test that if we create a hard link for a file F in some directory A, then move
```

Comment:
Can we change these to a step by step form
1. Create Hard link for File F in dir A
2. ... 

etc? It would be easier to read.